tests/installation/addon_products_sle.pm

```diff
@@ -42,7 +42,7 @@ sub handle_all_packages_medium {
     push @addons, 'we' if check_var('SLE_PRODUCT', 'sled') && !grep(/^we$/, @addons);
 
     # The legacy module is required if upgrade from previous version (bsc#1066338)
-    push @addons, 'legacy' if get_var('UPGRADE') && !grep(/^legacy$/, @addons);
+    push @addons, 'legacy' if get_var('UPGRADE') && !grep(/^legacy$/, @addons) && !check_var('SLE_PRODUCT', 'sled');
```

Comment:
Better to add a line to comment why legacy should not be selected and with bug number